src/cargo/core/features.rs

```diff
@@ -783,6 +784,7 @@ unstable_cli_options!(
     profile_rustflags: bool = ("Enable the `rustflags` option in profiles in .cargo/config.toml file"),
     public_dependency: bool = ("Respect a dependency's `public` field in Cargo.toml to control public/private dependencies"),
     publish_timeout: bool = ("Enable the `publish.timeout` key in .cargo/config.toml file"),
+    root_path: Option<PathBuf> = ("Set the root directory relative to which paths in diagnostics are printed (defaults to workspace root)"),
```

Comment:
FWIW I am not sure if it's *just* diagnostics, or also affects other paths rustc emits, like `file!()`.

Reply:
Yeah it does affect `file!()`, so I'll update the description